File: Jack.Business/ColaboradorCrianca.vb
Public Class ColaboradorCrianca
    Implements ICrud(Of Model.ColaboradorCrianca, Integer)

    Public Sub New()

    End Sub

#Region "Outros Métodos"

    Public Function ObterCriancasPorColaborador(intColaborador As Integer, intAno As Integer) As List(Of Model.ColaboradorCrianca)
        Dim oDados As Data.ColaboradorCrianca = Nothing
        Dim lstRetorno As IList(Of Model.ColaboradorCrianca) = Nothing

        Try
            oDados = New Data.ColaboradorCrianca
            lstRetorno = oDados.ObterCriancasPorColaborador(intColaborador, intAno)
        Catch ex As Exception
            lstRetorno = Nothing
            Throw ex
        Finally
            oDados = Nothing
        End Try

        Return lstRetorno

    End Function

#End Region

#Region "Crud"
    Public Function Delete(oTipo As Model.ColaboradorCrianca) As Boolean Implements ICrud(Of Model.ColaboradorCrianca, Integer).Delete
        Dim oDados As Data.ColaboradorCrianca = Nothing
        Dim blnRetorno As Boolean = False

        Try
            oDados = New Data.ColaboradorCrianca
            blnRetorno = oDados.Delete(oTipo)
        Catch ex As Exception
            blnRetorno = False
            Throw ex
        Finally
            oDados = Nothing
        End Try

        Return blnRetorno

    End Function

    Public Function Find(Identifier As Integer) As Model.ColaboradorCrianca Implements ICrud(Of Model.ColaboradorCrianca, Integer).Find

        Dim oDados As Data.ColaboradorCrianca = Nothing
        Dim lstRetorno As IList(Of Model.ColaboradorCrianca) = Nothing

        Try
            oDados = New Data.ColaboradorCrianca
            lstRetorno = oDados.LoadAll()
        Catch ex As Exception
            lstRetorno = Nothing
            Throw ex
        Finally
            oDados = Nothing
        End Try

        Return lstRetorno

    End Function

    Public Function Insert(oTipo As Model.ColaboradorCrianca) As Boolean Implements ICrud(Of Model.ColaboradorCrianca, Integer).Insert
        Dim oDados As Data.ColaboradorCrianca = Nothing
        Dim blnRetorno As Boolean = False

        Try
            oDados = New Data.ColaboradorCrianca
            blnRetorno = oDados.Insert(oTipo)
        Catch ex As Exception
            blnRetorno = False
            Throw ex
        Finally
            oDados = Nothing
        End Try

        Return blnRetorno
    End Function

    Public Function LoadAll() As IList(Of Model.ColaboradorCrianca) Implements ICrud(Of Model.ColaboradorCrianca, Integer).LoadAll
        Dim oDados As Data.ColaboradorCrianca = Nothing
        Dim lstRetorno As IList(Of Model.ColaboradorCrianca) = Nothing

        Try
            oDados = New Data.ColaboradorCrianca
            lstRetorno = oDados.LoadAll()
        Catch ex As Exception
            lstRetorno = Nothing
            Throw ex
        Finally
            oDados = Nothing
        End Try

        Return lstRetorno

    End Function

    Public Function Update(oTipo As Model.ColaboradorCrianca) As Boolean Implements ICrud(Of Model.ColaboradorCrianca, Integer).Update
        Dim oDados As Data.ColaboradorCrianca = Nothing
        Dim blnRetorno As Boolean = False

        Try
            oDados = New Data.ColaboradorCrianca
            blnRetorno = oDados.Update(oTipo)
        Catch ex As Exception
            blnRetorno = False
            Throw ex
        Finally
            oDados = Nothing
        End Try

        Return blnRetorno
    End Function
#End Region

End Class

File: Jack.Business/ColaboradorCrianca.cs
using System;
using System.Collections.Generic;

namespace Jack.Business
{
    public class ColaboradorCrianca : ICrud<Model.ColaboradorCrianca, int>
    {


        public ColaboradorCrianca()
        {
        }

        #region "Outros Métodos"

        public IList<Model.ColaboradorCrianca> ObterCriancasPorColaborador(int intColaborador, int intAno)
        {
            Data.ColaboradorCrianca oDados = null;
            IList<Model.ColaboradorCrianca> lstRetorno = null;

            try
            {
                oDados = new Data.ColaboradorCrianca();
                lstRetorno = oDados.ObterCriancasPorColaborador(intColaborador, intAno);
            }
            catch (Exception ex)
            {
                lstRetorno = null;
                throw ex;
            }
            finally
            {
                oDados = null;
            }

            return lstRetorno;

        }

        #endregion

        #region "Crud"
        public bool Delete(Model.ColaboradorCrianca oTipo)
        {
            Data.ColaboradorCrianca oDados = null;
            bool blnRetorno = false;

            try
            {
                oDados = new Data.ColaboradorCrianca();
                blnRetorno = oDados.Delete(oTipo);
            }
            catch (Exception ex)
            {
                blnRetorno = false;
                throw ex;
            }
            finally
            {
                oDados = null;
            }

            return blnRetorno;

        }

        public Model.ColaboradorCrianca Find(int Identifier)
        {

            Data.ColaboradorCrianca oDados = null;
            Model.ColaboradorCrianca oRetorno = null;

            try
            {
                oDados = new Data.ColaboradorCrianca();
                oRetorno = oDados.Find(Identifier);
            }
            catch (Exception ex)
            {
                oRetorno = null;
                throw ex;
            }
            finally
            {
                oDados = null;
            }

            return oRetorno;

        }

        public bool Insert(Model.ColaboradorCrianca oTipo)
        {
            Data.ColaboradorCrianca oDados = null;
            bool blnRetorno = false;

            try
            {
                oDados = new Data.ColaboradorCrianca();
                blnRetorno = oDados.Insert(oTipo);
            }
            catch (Exception ex)
            {
                blnRetorno = false;
                throw ex;
            }
            finally
            {
                oDados = null;
            }

            return blnRetorno;
        }

        public IList<Model.ColaboradorCrianca> LoadAll()
        {
            Data.ColaboradorCrianca oDados = null;
            IList<Model.ColaboradorCrianca> lstRetorno = null;

            try
            {
                oDados = new Data.ColaboradorCrianca();
                lstRetorno = oDados.LoadAll();
            }
            catch (Exception ex)
            {
                lstRetorno = null;
                throw ex;
            }
            finally
            {
                oDados = null;
            }

            return lstRetorno;

        }

        public bool Update(Model.ColaboradorCrianca oTipo)
        {
            Data.ColaboradorCrianca oDados = null;
            bool blnRetorno = false;

            try
            {
                oDados = new Data.ColaboradorCrianca();
                blnRetorno = oDados.Update(oTipo);
            }
            catch (Exception ex)
            {
                blnRetorno = false;
                throw ex;
            }
            finally
            {
                oDados = null;
            }

            return blnRetorno;
        }
        #endregion

    }
}
